src/halmos/__main__.py

```diff
@@ -1569,6 +1504,9 @@ def on_signal(signum, frame):
 
         contract_timer = NamedTimer("time")
 
+        # TODO: fix potential race conditions in the test-parallel mode
+        BuildOutMap().set_build_out_map(build_out_map)
+
```

Comment:
nice to see all tests in the test-parallel mode failed as expected :sweat_smile:

Reply:
i was wrong. the previous test failure was due to the singleton class being not shared across multi processes, not because of race conditions.

for the same reason, the trace pretty-printing feature doesn't work in the test-parallel mode.

Reply:
this particular issue has been fixed by having each process to have its own BuildOutMap, but it causes performance overhead in the test-parallel mode: https://github.com/a16z/halmos/pull/360#discussion_r1740491771

Reply:
offline discussion: let's get rid of test-parallel mode in future work